torch/nn/modules/module.py

```diff
@@ -2505,7 +2507,7 @@ def zero_grad(self, set_to_none: bool = True) -> None:
             set_to_none (bool): instead of setting to zero, set the grads to None.
                 See :meth:`torch.optim.Optimizer.zero_grad` for details.
         """
-        if getattr(self, '_is_replica', False):
+        if self._is_replica:
```

Comment:
i'm not sure if this is a safe change. a module can be reconstructed from a .pickle via torch.load() or via torchscript load.  In those cases, the module would not necessarily have the '_is_replica' property and this code would crash.

Reply:
That is fair... let me do some testing with `pickle` and `dill`.

Reply:
You are right... This is not a safe change. Great catch.